kucherov_sergeev/graph_generator.cpp

```diff
@@ -103,28 +162,85 @@ Graph GraphGenerator::generate() const {
   if (params_.depth() != 0) {
     graph.add_vertex();
     generate_grey_edges(graph);
```

Comment:
Вы внутри `generate_grey_edges` исчпользуете хитрую логику чтобы получить id корневой вершины, при чем используете там 0 как `magic number`. Много лучше будет передать туда реальные данные
```suggestion
    const auto root_id = graph.add_vertex();
    generate_grey_edges(graph, root_id);
```